hack/jenkins/e2e-runner.sh

```diff
@@ -207,6 +207,7 @@ function setup_gci_vars() {
 if running_in_docker; then
     record_command "${STAGE_PRE}" "download_gcloud" curl -fsSL --retry 3 --keepalive-time 2 -o "${WORKSPACE}/google-cloud-sdk.tar.gz" 'https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.tar.gz'
     install_google_cloud_sdk_tarball "${WORKSPACE}/google-cloud-sdk.tar.gz" /
+    gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
```

Comment:
Yeah, I agree with the reasoning behind needing to add the `activate-service-account` builder (while at the same time bemoaning our myriad of entry points :) )

I would like to be able to easily test how e2e-runner.sh will behave inside of docker when I am outside of docker